service/telemetry/internal/migration/v0.3.0.go

```diff
@@ -29,7 +29,10 @@ type TracesConfigV030 struct {
 
 func (c *TracesConfigV030) Unmarshal(conf *confmap.Conf) error {
 	if err := conf.Unmarshal(c); err != nil {
-		var v2TracesConfig tracesConfigV020
+		v2TracesConfig := tracesConfigV020{
+			Level:       c.Level,
+			Propagators: c.Propagators,
```

Comment:
I think you need to save initial values here.

Reply:
saved a copy of the values before unmarshaling